### ISSUE TITLE
RegisterOperationBlockStartAction inconsistent callback for properties

### ISSUE THREAD

**Issue by Evangelink** [dotnet/roslyn#46132]
For one analyzer, we have registered a symbol start action on `NamedType` on which we register an operation block start action. Thanks to some false-positive being reported, I have noticed that a reaodnly auto property doesn't get any callback while the same readonly auto property with an attribute on the getter now gets a callback.

I don't know if it's a bug or something intended but that seems weird to me to have such a different behavior. I would expect to have a callback on both cases.

```csharp
public sealed class C1
{
    public C1(string value) => Value = value; // Callback

    public string Value { get; } // No callback
}

public sealed class C2
{
    public C2(string value) => Value = value; // Callback

    public string Value
    {
        [System.Diagnostics.DebuggerStepThrough]
        get; // Callback
    }
}
```

Related side question, from the symbol or operation context, is it possible to know this is an auto-property?

**Comment by CyrusNajmabadi** [comment]
> Related side question, from the symbol or operation context, is it possible to know this is an auto-property?

Yes.  You can get the containing type, and see if any IFieldSymbols have this property as their associatedsymbol.  @jnm2 is working on an API change that makes that simpler i believe.

**Comment by Evangelink** [comment]
I have used a different technique: `blockStartContext.OperationBlocks.Where(op => op.Kind != OperationKind.None).Any()` which seems to be working well. Do you think that's too risky?
